urllib3/util/retry.py

```diff
@@ -169,10 +179,13 @@ def get_backoff_time(self):
 
         :rtype: float
         """
-        if self._observed_errors <= 1:
+        # We want to consider only the last consecutive errors sequence (Ignore redirects).
+        consecutive_errors_len = len(list(takewhile(lambda x: x.redirect_location is None,
```

Comment:
Not necessarily clearer but could also write this as:

```python
consecutive_errors_len = next(i for i, h in enumerate(reversed(self.history)) if not h.redirect_location)
```

At least no lambdas and itertools, but either way is fine. :) Just a nit.

Also any thoughts if it would it make more sense to look at `.error` or `.status` instead?